### ISSUE TITLE
DL3025 regression using JSON notation with escaped quotes in strings

### ISSUE THREAD

**Issue by zechris** [hadolint/hadolint#802]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
It should handle escaped quotes `\"` in strings correctly, like v2.8.0 did.
(Perhaps this was introduced in the [language-docker 10.4.1->10.4.2](https://github.com/hadolint/language-docker/compare/10.4.1...10.4.2)?)

### Actual behavior
This should lint correctly, like v2.8.0 did.

### Steps to reproduce the behavior

**Output of `hadolint --version` 
```
  Haskell Dockerfile Linter 2.9.1-no-git
```

```bash
❯ hadolint Dockerfile
Dockerfile:2 DL3025 warning: Use arguments JSON notation for CMD and ENTRYPOINT arguments
```

**Dockerfile** (if relevant)

```Dockerfile
FROM debian:8
ENTRYPOINT ["sbt", "set reStart / mainClass := Some(\"Main\");~reStart"]
```

**Additional environment details (OS, stack version, etc.)**
